flocker/control/test/test_protocol.py

```diff
@@ -732,6 +732,31 @@ def test_configuration_change(self):
             dict(configuration=agent.desired, state=agent.actual),
         )
 
+    def test_unexpected_exception_raised(self):
+        """
+        A configuration change results in connected protocols being notified
+        of new cluster status.
+        """
+        agent = FakeAgent()
+        client = AgentAMP(Clock(), agent)
+        service = build_control_amp_service(self)
+
+        def raise_unexpected_exception(self,
+                                       commandType=None,
+                                       *a, **kw):
+            raise Exception("I'm an unexpected exception")
+
+        service.startService()
+        server = LoopbackAMPClient(client.locator)
+        self.patch(server,
+                   "callRemote",
+                   raise_unexpected_exception
+                   )
+
+        service.connected(server)
+
+        service.configuration_service.save(TEST_DEPLOYMENT)
```

Comment:
This does test that the exception isn't raised, but it doesn't test the expected behavior which is that if connection A gets exception raised that doesn't impact updates to connection B. There should be a test case for that in the case where callRemote returns a `Failure`, so a variant of that is probably what you want (perhaps with shared code for most of it?).